tests/distributions/test_negative_binomial.py

```diff
@@ -0,0 +1,54 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import numpy as np
+import pyro.distributions as dist
+import pytest
+import torch
+
+from cellarium.ml.distributions import NegativeBinomial
+
+
+@pytest.mark.parametrize("logits_shape", [(), (2,), (3, 2)])
+@pytest.mark.parametrize("total_counts_shape", [(), (2,), (3, 2)])
+def test_negative_binomial(logits_shape: torch.Size, total_counts_shape: torch.Size) -> None:
+    logits = torch.randn(logits_shape)
+    total_counts = torch.rand(total_counts_shape) * 10
```

Comment:
Maybe it would also make sense to hit some edge cases, like the reversion to Poisson when $\theta \rightarrow \inf$, and things like small $\mu$ and extremely large $\theta$.

Reply:
Let's make another test and sweep `mu` in `[1e-4, ..., 1e3]` and `theta` in `[1e-2, ..., 1e3]` calculate the total probability (via logsumexp over a range of `value` in `[0, ..., 1e3 * mu]`) and assert that it is close to `1.0` with a tolerance of `1e-5` using `float32`.

Reply:
Added test, however, it only passes if I set `atol=5e-4`

Reply:
Like, is there a plot of y-axis `THETA_THRESHOLD_STIRLING_SWITCH`, x-axis `atol` that could be created?

Reply:
> Does the necessary `atol` value change if you increase `THETA_THRESHOLD_STIRLING_SWITCH` to like 500 or 1000 ?

Just tried it, it didn't. My understanding from that wiki article that this approximation breaks down only for very small theta.